mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -534,8 +577,15 @@ class Word2VecModel private[spark] (
     // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
     var ind = 0
+    var vecNorm = 1f
+    if (norm) {
```

Comment:
As the comment says it's not normalized just because it doesn't affect ordering. However, the API promises a "cosine similarity", and that's not what it is. Since it need only be applied to the result, I think it should be an actual cosine similarity. I do not think it should have a flag, no.